hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledTaskDescriptor.java

```diff
@@ -214,6 +223,23 @@ public void readData(ObjectDataInput in)
         resultRef.set((ScheduledTaskResult) in.readObject());
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ScheduledTaskDescriptor that = (ScheduledTaskDescriptor) o;
+        return  (definition == that.definition) || (definition != null && definition.equals(that.definition));
+    }
+
+    @Override
+    public int hashCode() {
+        return Arrays.hashCode(new TaskDefinition[]{definition});
```

Comment:
I don't know whether this `hashCode` method is called frequently, but using ideas default template to generate it can help to remove unneeded `TaskDefinition` object creation

Reply:
:+1: for re-creation via IntelliJ for `equals()` + `hashCode()`. I would also recommend to write a test with this fancy library Vassilis integrated. You need to make the methods `final`. For a test stub I would have a look at the config class tests.

Reply:
This *is* one of the IntelliJ templates, (a new one). I chose this one, because it reduces the clatter and the annoying findbugs annotations.

Reply:
if it is not a frequently called method, i 'm ok with current form of it.